src/document/docUtil.ts

```diff
@@ -0,0 +1,53 @@
+import { SelectionItem } from '../store';
+
+/**
+ * Finds the closest ancestor DOM element with the specified class.
+ */
+export function findClosestElWithClass(element: Element | null, className: string): Element | null {
+    for (let el = element; el && el !== ((document as unknown) as Element); el = el.parentNode as Element | null) {
```

Comment:
We should not need this much explicit casting. Can you take a look at `getScrollParent` in `useAutoScroll/util`?